### PR TITLE
BUG: GroupBy with subquery.

### PR DESCRIPTION
**Pull request by igor-tkachev**
The following code 

```c#
from pmp in
(
	from pmp  in db.Child
	group pmp by pmp.ParentID into g
	select g.Key
)
from pmp1 in db.Child
select new { pmp1.ChildID }
```

generates wrong SQL

```sql
SELECT
	[pmp1].[ChildID]
FROM
	[Child] [pmp],
	[Child] [pmp1]
GROUP BY
	[pmp].[ParentID]
```

should be 

```sql
SELECT
	[pmp1].[ChildID]
FROM
	(
		SELECT
			[t1].[ParentID]
		FROM
			[Child] [t1]
		GROUP BY
			[t1].[ParentID]
	) [pmp],
	[Child] [pmp1]
```

### CODE REVIEW

File: Source/LinqToDB/SqlQuery/SelectQueryOptimizer.cs [hunk 1]
```diff
@@ -1508,21 +1508,26 @@ void OptimizeSubQueries(bool isApplySupported)
 		{
 			CorrectCrossJoinQuery(_selectQuery);
 
-			for (var i = 0; i < _selectQuery.From.Tables.Count; i++)
+			var tableSources = _selectQuery.From.Tables;
+
+			for (var i = 0; i < tableSources.Count; i++)
 			{
-				var table = OptimizeSubQuery(_selectQuery, _selectQuery.From.Tables[i], true, false, isApplySupported, true, null);
+				if (tableSources.Count > 1 && tableSources[i] is { Source: SelectQuery { GroupBy.IsEmpty: false }})
```
Comment (igor-tkachev) [inline Source/LinqToDB/SqlQuery/SelectQueryOptimizer.cs:1515]: Fix.

**Comment by sdanyliv** [comment]
I would suggest to use `AassertQuery` to validate grouping result.

**Review by MaceWindu — DISMISSED** [review]
Could you specify what was broken? And I will agree with @sdanyliv - we should have assertions as from time to time we discover tests that doesn't fail but generated sql doesn't match intended logic

**Review by MaceWindu — APPROVED** [review]

**Pull request merged**